server/services/lobby_service.go

```diff
@@ -49,19 +61,47 @@ func (l *LobbyServiceImpl) CreateLobby() *Lobby {
 	return newLobby
 }
 
-// GetLobby returns a map of lobbies currently active
-func (l *LobbyServiceImpl) GetLobby(lobbyID string) (*Lobby, bool) {
+// IfLobbyExists calls the given function if the lobby is found, returning a copy of the
+// modified lobby if it was found. A bool will be returned indicating if the lobby was found
```

Comment:
This function now accepts a `LobbyModFunc` which is a function that takes a lobby pointer as an argument. This is the provided thread-safe way of modifying lobbies. See the usages for examples of how this looks, and feel free to ask me questions in case this doesn't make sense.